README.md

```diff
@@ -1,10 +1,458 @@
+
+## API
+
+### `cloudflare`
+
+The `cloudflare` plugin should be included in the Vite `plugins` array:
+
+```ts
+// vite.config.ts
+
+import { defineConfig } from 'vite'
+import { cloudflare } from '@cloudflare/vite-plugin'
+
+export default defineConfig({
+  plugins: [cloudflare()],
+})
+```
+
+It accepts an optional `PluginConfig` parameter.
+
+### `interface PluginConfig`
+
+- `configPath?: string`
+
+  An optional path to your Worker config file.
+  By default, a `wrangler.toml`, `wrangler.json`, or `wrangler.jsonc` file in the root of your application will be used as the Worker config.
+
+- `viteEnvironment?: { name?: string }`
+
+  Optional Vite environment options.
+  By default, the environment name is the Worker name with `-` characters replaced with `_`.
+  Setting the name here will override this.
+
+- `persistState?: boolean | { path: string }`
+
+  An optional override for state persistence.
+  By default, state is persisted to `.wrangler/state` in a `v3` subdirectory.
+  A custom `path` can be provided or, alternatively, persistence can be disabled by setting the value to `false`.
+
+- `auxiliaryWorkers?: Array<AuxiliaryWorkerConfig>`
+
+  An optional array of auxiliary workers.
+  You can use [service bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/service-bindings/) to call auxiliary workers from your main (entry) Worker.
+  All requests are routed through your entry Worker.
+  During the build, each Worker is output to a separate subdirectory of `dist`.
+
+> [!NOTE]
+> When running `wrangler deploy`, only your main (entry) Worker will be deployed.
+> If using multiple Workers, each must be deployed individually.
+> You can inspect the `dist` directory and then run `wrangler deploy -c path-to-worker-output-config` for each.
+
+### `interface AuxiliaryWorkerConfig`
+
+- `configPath: string`
+
+  A required path to your Worker config file.
+
+- `viteEnvironment?: { name?: string }`
+
+  Optional Vite environment options.
+  By default, the environment name is the Worker name with `-` characters replaced with `_`.
+  Setting the name here will override this.
+
+## Worker environments
+
+A Worker config file may contain configuration for multiple [environments](https://developers.cloudflare.com/workers/wrangler/environments/).
+With the Cloudflare Vite plugin, you select an environment at dev or build time by providing the `CLOUDFLARE_ENV` environment variable.
+Consider the following example `wrangler.toml` file:
+
+```toml
+# wrangler.toml
+
+name = "my-worker"
+compatibility_date = "2024-12-30"
+main = "./src/index.ts"
+
+vars = { MY_VAR = "Top-level var" }
+
+[env.staging]
+vars = { MY_VAR = "Staging var" }
+
+[env.production]
+vars = { MY_VAR = "Production var" }
+```
+
+If you run `CLOUDFLARE_ENV=production vite build` then the output `wrangler.json` file generated by the build will be a flattened configuration for the 'production' environment.
```

Comment:
We does this only specify `vite build` without any mention of `vite dev`?

Reply:
I couldn't find a way of including both without it becoming overly long and confusing (there's no output `wrangler.json` in dev for example). I have called out either side of this paragraph that it can be applied at dev and build time. I can have another think.

Reply:
yeah I can see that you did mention "dev or build time", maybe that's good enough? 🤔 

In any case I'm fine with the current paragraph, we can always improve it later if needed

Reply:
I've added an additional paragraph - [886873f](https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/119/commits/886873f6fa7528620dc5cdc83d64ef689a18a5ae).